Feature/add ring doorbell

graham33 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Required for the Home Assistant Ring integration.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

graham33 committed:
python3Packages.ring_doorbell: init at 0.7.0

graham33 commented:
Actually, it looks like Home Assistant breaks with 0.7.0 (looks like an API change, I have previously been using 0.6.2 successfully).  I'll downgrade this since it seems better to commit something that works with Home Assistant.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 114181` at 50230a48 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.ring_doorbell</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/15eca070955845e84862814c63efab18">python37Packages.ring_doorbell</a></li></li>
    <li><a href="https://gist.github.com/27b8b2c1f59030a68125861cd246801b">python38Packages.ring_doorbell</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl2` is a deprecated license, check if project uses `lgpl2Plus` or `lgpl2Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/ring_doorbell/default.nix:38:5:
```
   |
38 |     license = licenses.lgpl2;
   |     ^
```
</li>
  </ul>
</details>

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -6897,6 +6897,8 @@ in {
 
   rig = callPackage ../development/python-modules/rig { };
 
+  ring_doorbell = callPackage ../development/python-modules/ring_doorbell { };
```
```suggestion
  ring-doorbell = callPackage ../development/python-modules/ring_doorbell { };
```
https://nixos.org/manual/nixpkgs/stable/#contributing

I think this also applies to `pname`.

graham33 replied:
I've changed this, thanks.  I presume it should also apply to the directory name (which I changed too)?  I had to override the `pname` to use underscores for fetchPyPI.

fabaff commented on pkgs/development/python-modules/ring_doorbell/default.nix:
```diff
@@ -0,0 +1,41 @@
+  meta = with lib; {
+    homepage = "https://github.com/tchellomello/python-ring-doorbell";
+    description = "A Python library to communicate with Ring Door Bell (https://ring.com/)";
+    license = licenses.lgpl2;
```
The license tag in the `setup.py` file says [LGPLv2+](https://github.com/tchellomello/python-ring-doorbell/blob/master/setup.py#L23).

```suggestion
    license = licenses.lgpl2Plus;
```

graham33 replied:
It actually says LGPL v3+, but the actual LICENSE file is v2 so I think what you suggest makes sense.

dotlambda replied:
You should file an upstream pull request to fix that.

graham33 replied:
I filed an issue since I'm not sure which version they're actually intending to use: https://github.com/tchellomello/python-ring-doorbell/issues/241.

fabaff replied:
> It actually says LGPL v3+...

Right

graham33 replied:
Upstream has fixed their LICENSE file, so updated to `lgpl3Plus`.

fabaff commented on pkgs/development/python-modules/ring_doorbell/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+buildPythonPackage rec {
+  pname = "ring_doorbell";
+  version = "0.6.2";
```
The latest release is [0.7.0](https://github.com/tchellomello/python-ring-doorbell/releases/tag/0.7.0).

graham33 replied:
(See comment above, I initially tried this version but it doesn't appear to work with current latest Home Assistant)

fabaff replied:
Home Assistant is pinning the releases. The problem is that bots could pick it up as outdated and then it will be updated anyways.

With my "Home Assistant hat" on:

I try to keep the requirements of Home Assistant as up-to-date as possible but I don't have access to every IoT device or service that is integrated. If you have a Ring device then please consider to update the [`ring` integration](https://github.com/home-assistant/core/blob/dev/homeassistant/components/ring/manifest.json#L5) to 0.7.0. This way all Ring users could benefit. Thanks.

graham33 replied:
Looks like the issue is acknowledged as a bug upstream (https://github.com/tchellomello/python-ring-doorbell/issues/240), I don't think they intended a breaking change, so presumably it'll be fixed in a new release at some point.  From my point of view it seems best to stick on 0.6.2 until that fix comes out, rather than merge a known-buggy version.

I'm not very familiar with the upgrade bot process.  Assuming any changes are viewed by a human, perhaps a TODO comment iwould suffice?

dotlambda replied:
I think we should package version 7.0.0 and override the version inside of `pkgs/servers/home-assistant/default.nix`.

> Assuming any changes are viewed by a human, perhaps a TODO comment iwould suffice?

They are not, only in case of a build failure.

graham33 replied:
I see, I'm not familiar with how this works so thanks for taking the time to explain!  I've made the suggested changes.

fabaff commented on pkgs/development/python-modules/ring_doorbell/default.nix:
```diff
@@ -0,0 +1,41 @@
+    pytz
+    requests
+    requests_oauthlib
+  ];
```
According to `install_requires` is [`oauthlib`](https://github.com/tchellomello/python-ring-doorbell/blob/master/setup.py#L28) a requirement. `oauthlib` should be mentioned even if `requests_oauthlib` is pulling it in. This avoids issues if `requests_oauthlib` drops `oauthlib`.

graham33 replied:
Fixed, thanks.

graham33 committed:
home-assistant: add ring-doorbell dependency

For the ring integration.  This is currently pinned back to 0.6.2 to avoid an
upstream bug in 0.7.0.

dotlambda commented on pkgs/development/python-modules/ring-doorbell/default.nix:
```diff
@@ -0,0 +1,44 @@
+  meta = with lib; {
+    homepage = "https://github.com/tchellomello/python-ring-doorbell";
+    description = "A Python library to communicate with Ring Door Bell (https://ring.com/)";
+    license = licenses.lgpl2Plus;
```
```suggestion
    license = licenses.lgpl3Plus;
```

fabaff commented:
Result of `nixpkgs-review pr 114181` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.ring-doorbell</li>
    <li>python38Packages.ring-doorbell</li>
    <li>python39Packages.ring-doorbell</li>
  </ul>
</details>
